scene/surface-placement/src/main/java/com/esri/samples/surface_placement/SurfacePlacementController.java

```diff
@@ -0,0 +1,203 @@
+
+      // create a text symbol for each surface placement type
+      TextSymbol drapedBillboardedText =
+          new TextSymbol(14, "DRAPED BILLBOARDED", 0xFF0000FF, TextSymbol.HorizontalAlignment.LEFT, TextSymbol.VerticalAlignment.MIDDLE);
+      drapedBillboardedText.setOffsetX(20);
+
+      TextSymbol drapedFlatText =
+          new TextSymbol(14, "DRAPED FLAT", 0xFF0000FF, TextSymbol.HorizontalAlignment.LEFT, TextSymbol.VerticalAlignment.MIDDLE);
+      drapedFlatText.setOffsetX(20);
+
+      TextSymbol relativeText =
+          new TextSymbol(14, "RELATIVE", 0xFF0000FF, TextSymbol.HorizontalAlignment.LEFT, TextSymbol.VerticalAlignment.MIDDLE);
+      relativeText.setOffsetX(20);
+
+      TextSymbol absoluteText =
+          new TextSymbol(14, "ABSOLUTE", 0xFF0000FF, TextSymbol.HorizontalAlignment.LEFT, TextSymbol.VerticalAlignment.MIDDLE);
+      absoluteText.setOffsetX(20);
+
+      TextSymbol relativeToSceneText =
+        new TextSymbol(14, "RELATIVE TO SCENE", 0xFF0000FF, TextSymbol.HorizontalAlignment.RIGHT, TextSymbol.VerticalAlignment.MIDDLE);
+      relativeToSceneText.setOffsetX(-20);
+
+      // add the point graphic and text graphic to the corresponding graphics overlay
+      drapedBillboardedOverlay.getGraphics().add(new Graphic(surfaceRelatedPoint, triangleSymbol));
+      drapedBillboardedOverlay.getGraphics().add(new Graphic(surfaceRelatedPoint, drapedBillboardedText));
+
+      drapedFlatOverlay.getGraphics().add(new Graphic(surfaceRelatedPoint, triangleSymbol));
+      drapedFlatOverlay.getGraphics().add(new Graphic(surfaceRelatedPoint, drapedFlatText));
+
+      relativeOverlay.getGraphics().add(new Graphic(surfaceRelatedPoint, triangleSymbol));
+      relativeOverlay.getGraphics().add(new Graphic(surfaceRelatedPoint, relativeText));
+
+      absoluteOverlay.getGraphics().add(new Graphic(surfaceRelatedPoint, triangleSymbol));
+      absoluteOverlay.getGraphics().add(new Graphic(surfaceRelatedPoint, absoluteText));
+
+      relativeToSceneOverlay.getGraphics().add(new Graphic(sceneRelatedPoint, triangleSymbol));
+      relativeToSceneOverlay.getGraphics().add(new Graphic(sceneRelatedPoint, relativeToSceneText));
+
+      // add graphics overlays to the scene view
+      sceneView.getGraphicsOverlays().addAll(Arrays.asList(drapedBillboardedOverlay, relativeOverlay, absoluteOverlay, relativeToSceneOverlay));
+
+      // link up the ToggleGroup RadioButtons to the corresponding draped overlay
+      drapedBillboardedRadioButton.setUserData(drapedBillboardedOverlay);
+      drapedFlatRadioButton.setUserData(drapedFlatOverlay);
+
+      // listener for toggles between billboarded and flat draped surface placement modes on the JavaFX ToggleGroup
+      toggleGroup.selectedToggleProperty().addListener((observableValue, oldToggle, newToggle) -> {
+        if (toggleGroup.getSelectedToggle() != null) {
+          sceneView.getGraphicsOverlays().remove(oldToggle.getUserData());
```

Comment:
Maybe it'll be safer to `clear` all the graphics overlays here, so that we're essentally just zapping everything that's visible, and not relying on the `oldToggle` property.

```suggestion
          sceneView.getGraphicsOverlays().clear();
```

Reply:
I tried with clear but it then removes the other overlays too. Shall I leave with the `oldToggle`, or I could do an `addAll()` on the next line that includes the other graphics.

Reply:
You're absolutely right, I made a mistake there! Forgot that there's a bunch of other GraphicsOverlays, not just switching between 2. It was correct the way it was before, sorry!